x-pack/plugins/lists/common/schemas/request/create_endpoint_list_item_schema.test.ts

```diff
@@ -0,0 +1,216 @@
+    };
+    const decoded = createEndpointListItemSchema.decode(inputPayload);
+    const checked = exactCheck(inputPayload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual(['invalid keys "namespace_type"']);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should validate an undefined for "meta" but strip it out and generate a correct body not counting the auto generated uuid', () => {
+    const payload = getCreateEndpointListItemSchemaMock();
+    const outputPayload = getCreateEndpointListItemSchemaMock();
+    delete payload.meta;
+    delete outputPayload.meta;
+    const decoded = createEndpointListItemSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    delete (message.schema as CreateEndpointListItemSchema).item_id;
+    expect(getPaths(left(message.errors))).toEqual([]);
+    expect(message.schema).toEqual(outputPayload);
+  });
+
+  test('it should validate an undefined for "comments" but return an array and generate a correct body not counting the auto generated uuid', () => {
+    const inputPayload = getCreateEndpointListItemSchemaMock();
+    const outputPayload = getCreateEndpointListItemSchemaMock();
+    delete inputPayload.comments;
+    outputPayload.comments = [];
+    const decoded = createEndpointListItemSchema.decode(inputPayload);
+    const checked = exactCheck(inputPayload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    delete (message.schema as CreateEndpointListItemSchema).item_id;
+    expect(getPaths(left(message.errors))).toEqual([]);
+    expect(message.schema).toEqual(outputPayload);
+  });
+
+  test('it should validate "comments" array', () => {
+    const inputPayload = {
+      ...getCreateEndpointListItemSchemaMock(),
+      comments: getCreateCommentsArrayMock(),
+    };
+    const decoded = createEndpointListItemSchema.decode(inputPayload);
+    const checked = exactCheck(inputPayload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    delete (message.schema as CreateEndpointListItemSchema).item_id;
+    expect(getPaths(left(message.errors))).toEqual([]);
+    expect(message.schema).toEqual(inputPayload);
+  });
+
+  test('it should NOT validate "comments" with "created_at" or "created_by" values', () => {
+    const inputPayload: Omit<CreateEndpointListItemSchema, 'comments'> & {
+      comments?: CommentsArray;
+    } = {
+      ...getCreateEndpointListItemSchemaMock(),
+      comments: [getCommentsMock()],
+    };
+    const decoded = createEndpointListItemSchema.decode(inputPayload);
+    const checked = exactCheck(inputPayload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual(['invalid keys "created_at,created_by"']);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should validate an undefined for "entries" but return an array', () => {
```

Comment:
This is more just thinking aloud - do you think we should require items to have a non-empty array value for `entries`? Not sure I see any use for creating an item without entries. Right now in the UI, I'm stripping such items out and not displaying them.

Reply:
Probably be better to go strict than non-strict. It was simpler to do it with Joi than io-ts. I am finding we end up doing more validation stuff in the REST endpoints with io-ts which introduces 😢 sadness and complexity.

Reply:
oh no, not 😢  sadness! Yea, I agree. Io-ts is cool in a lot of ways, but there is a lot more complexity being added just to validate.